src/adapters/diksha/quml.adapter.ts

```diff
@@ -0,0 +1,130 @@
+import { Injectable } from "@nestjs/common";
+import { HttpService } from "@nestjs/axios";
+import { SuccessResponse } from "src/success-response";
+
+import { QuestionDto } from "src/Question/dto/question.dto";
+
+@Injectable()
+export class QumlQuestionService {
+  constructor(private httpService: HttpService) {}
+
+  public async getAllQuestions(
+    questionType: string,
+    subject: string,
+    language: string,
+    medium: string
+  ) {
+    var axios = require("axios");
+    var data = {
+      request: {
+        filters: {
+          objectType: "Question",
+          status: ["Live"],
+          qType: questionType,
+          subject: subject,
+          language: language,
+          medium: medium,
+        },
+      },
+    };
+    var config = {
+      method: "post",
+      url: "https://vdn.diksha.gov.in/action/composite/v3/search",
+      headers: {
+        "Content-Type": "application/json",
+      },
+      data: data,
+    };
+
+    const response = await axios(config);
+    const responseData = response.data.result.Question;
+
+    let arrayIds = responseData.map((e: any) => {
+      return e.identifier;
+    });
+
+    let questionArray = [];
+    for (let value of arrayIds) {
+      let config = {
+        method: "get",
+        url: `https://vdn.diksha.gov.in/action/question/v1/read/${value}?fields=body,qType,answer,responseDeclaration,name,solutions,editorState,media,name,board,medium,gradeLevel,subject,topic,learningOutcome,marks`,
```

Comment:
Move base URL to a class constant. 

Also, when testing you may need to connect to dev instance of diksha, how will you enable that ?

Reply:
sir for read the questions authentication not required

Reply:
It's not about authentication. The base URL of diksha will change per environment (eg: staging.diksha.gov.i), how will that be handled ?